### PR TITLE
Newfile protocol

### PR DESCRIPTION
**Pull request by mehdibalouchi**

### CODE REVIEW

File: protocols/newFile/constant.go [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package newFile
+
+const PROTOCOL = "/fx/new-file/0.1.0"
```
Comment (masih) [inline protocols/newFile/constant.go:3]: In go constants follow a simple CamelCase naming

File: protocols/newFile/handlers.go [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+	"github.com/golang/protobuf/proto"
+	"github.com/libp2p/go-libp2p-core/network"
+)
+
+// Root handler for newFile protocol
+// It reads 2 bytes from the stream which determines the size of the request messeage
+// It then reads the request message and passes the stream to the appropriate handler
+func Handle(ctx context.Context, api fxiface.CoreAPI, ds drive.DriveStore, s network.Stream) error {
+	fmt.Println("handling")
```
Comment (masih) [inline protocols/newFile/handlers.go:21]: Use logging library instead. Search for similar lines across the PR and replace as needed

File: protocols/newFile/protos/newFile.proto [hunk 3]
```diff
@@ -0,0 +1,52 @@
+syntax = "proto3";
+option go_package = "github.com/functionland/go-fula/protocols/newFile";
+package newFile;
```
Comment (masih) [inline protocols/newFile/protos/newFile.proto:3]: newFile seems like an odd package name; how about `file`?

**Review by masih — APPROVED** [review]

File: protocols/newFile/protos/newFile.proto (hunk 3 above)
Comment (mehdibalouchi) [reply protocols/newFile/protos/newFile.proto:3]: I will change the name to `file` and remove the old package after I finished the tests for the `mobile` package.

File: protocols/newFile/requests.go [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+
+func sendRequest(req *FSRequest, s network.Stream) error {
+	reqMsg, err := proto.Marshal(req)
+	if err != nil {
+		return err
+	}
+
+	// @TODO define the maximum bytes needed for sending the length of the req
+	sbuf := make([]byte, 2)
+	binary.LittleEndian.PutUint16(sbuf, uint16(len(reqMsg)))
```
[the remark this answers is not in the document]
Comment (mehdibalouchi) [reply protocols/newFile/requests.go:23]: I don't know, that's why I put TODO

[the remark this answers is not in the document]
Comment (masih) [reply protocols/newFile/requests.go:23]: If we don't know yet, why include it at all?

**Pull request merged**